go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -254,13 +254,11 @@ func (n *NodeController) AddNode(node *kapi.Node) error {
 	klog.Info("Add Node ", node.Name)
 	var err error
 	if n.drIP == nil {
-		subnet, err := getLocalNodeSubnet(n.nodeName)
-		if err != nil {
-			return err
+		hybridOverlayIFAddress, ok := node.Annotations[hotypes.HybridOverlayPortAddress]
+		if !ok {
+			return fmt.Errorf("hybrid overlay not initialized on %s, it was not assigned an interface address", node.Name)
 		}
-		// n.drIP is always 3rd address in the subnet
-		hybridOverlayIfAddr := util.GetNodeHybridOverlayIfAddr(subnet)
-		n.drIP = hybridOverlayIfAddr.IP
+		n.drIP = net.ParseIP(hybridOverlayIFAddress)
```

Comment:
NIT: may want to return an error if net.ParseIP returns nil

Reply:
done